### PR TITLE
add some check to operator.run

### PR DESCRIPTION
**Pull request by jacquesqiao**
fix: https://github.com/PaddlePaddle/Paddle/issues/4545

### CODE REVIEW

File: paddle/framework/operator.h [hunk 1]
```diff
@@ -478,9 +478,20 @@ class OperatorWithKernel : public OperatorBase {
     this->InferShape(&infer_shape_ctx);
 
     ExecutionContext ctx(*this, scope, dev_ctx);
-    auto& opKernel = AllOpKernels().at(type_).at(
-        OpKernelKey(IndicateDataType(ctx), dev_ctx));
-    opKernel->Compute(ctx);
+    PADDLE_ENFORCE_GT(AllOpKernels().count(type_), 0, "op[%s] has no kernel",
```
Comment (reyoung) [inline paddle/framework/operator.h]: Do not use `count` to check contains or not.

File: paddle/framework/operator.h [hunk 2]
```diff
@@ -478,9 +478,20 @@ class OperatorWithKernel : public OperatorBase {
...
+                      type_);
+    OpKernelMap& kernels = AllOpKernels().at(type_);
+    DataType data_type = IndicateDataType(ctx);
+    auto kernel_key = OpKernelKey(data_type, dev_ctx);
+    if (kernels.count(kernel_key) == 0) {
+      const ::google::protobuf::EnumDescriptor* descriptor =
+          DataType_descriptor();
+      std::string data_type_str =
+          descriptor->FindValueByNumber(data_type)->name();
+      PADDLE_THROW("op[%s] has no kernel with data_type[%s]", type_,
```
Comment (reyoung) [inline paddle/framework/operator.h]: Do not use reflection here. Since https://github.com/PaddlePaddle/Paddle/issues/4542

Comment (jacquesqiao) [reply paddle/framework/operator.h]: removed

File: paddle/framework/operator.h (hunk 1 above)
Comment (jacquesqiao) [reply paddle/framework/operator.h]: use find instead, thanks for pointing out

**Pull request merged**